conans/client/graph/graph_binaries.py

```diff
@@ -451,3 +455,14 @@ def reevaluate_node(self, node, remotes, build_mode, update):
         output.info("Binary for updated ID from: %s" % node.binary)
         if node.binary == BINARY_BUILD:
             output.info("Binary for the updated ID has to be built")
+
+
+def _call_validate_build(node):
+    conanfile = node.conanfile
+    if hasattr(conanfile, "validate_build") and callable(conanfile.validate_build):
+        with conanfile_exception_formatter(str(conanfile), "validate_build"):
+            try:
+                conanfile.validate_build()
+            except ConanInvalidConfiguration as e:
+                conanfile.info.invalid = str(e)
```

Comment:
This is weird. If it cannot be built, it doesn't depend on the ``info`` for ``package_id``. It is a property of the ``conanfile.cant_build = str(e)``. Overriding ``info.invalid`` can destroy information of the potential valid/invalid of the ``package_id``